src/polymer/polymer-element.ts

```diff
@@ -384,6 +316,132 @@ function inheritFrom(element: PolymerElement, superElement: PolymerElement) {
+    scannedElement: ScannedElement,
+    document: Document) {
+  if (scannedElement.superClass &&
+      scannedElement.superClass.identifier !== 'HTMLElement') {
+    const superElements =
+        document.getById('element', scannedElement.superClass.identifier, {
+          externalPackages: true,
+          imported: true,
+        });
+    if (superElements.size === 1) {
+      const superElement = superElements.values().next().value;
+      if (!superElement.kinds.has('polymer-element')) {
+        element.warnings.push({
+          message:
+              `A Polymer element can\'t extend from a non-Polymer element: ${scannedElement
+                  .superClass.identifier}`,
+          severity: Severity.ERROR,
+          code: 'unknown-superclass',
+          sourceRange: scannedElement.superClass.sourceRange!,
+        });
+      } else {
+        inheritFrom(element, superElement as PolymerElement);
+      }
+    } else {
+      if (superElements.size === 0) {
+        element.warnings.push({
+          message: `Unable to resolve superclass ${scannedElement.superClass
+                       .identifier}`,
+          severity: Severity.ERROR,
+          code: 'unknown-superclass',
+          sourceRange: scannedElement.superClass.sourceRange!,
+        });
+      } else {
+        element.warnings.push({
+          message: `Multiple superclasses found for ${scannedElement.superClass
+                       .identifier}`,
+          severity: Severity.ERROR,
+          code: 'unknown-superclass',
+          sourceRange: scannedElement.superClass.sourceRange!,
+        });
+      }
+    }
+  }
+}
+
+function applyMixins(
+    element: PolymerElement,
+    scannedElement: ScannedElement,
+    document: Document) {
+  for (const scannedMixinReference of scannedElement.mixins) {
+    const mixinReference = scannedMixinReference.resolve(document);
+    element.mixins.push(mixinReference);
+    const mixins =
+        document.getById('element-mixin', mixinReference.identifier, {
+          externalPackages: true,
+          imported: true,
+        });
+    if (mixins.size === 0) {
+      element.warnings.push({
+        message:
+            `@mixes reference not found ${scannedMixinReference.identifier}`,
+        severity: Severity.ERROR,
+        code: 'mixes-reference-not-found',
+        sourceRange: scannedMixinReference.sourceRange!,
+      });
+      continue;
+    } else if (mixins.size > 1) {
+      element.warnings.push({
+        message:
+            `@mixes reference, multiple mixins found ${scannedMixinReference
+                .identifier}`,
+        severity: Severity.ERROR,
+        code: 'mixes-reference-not-found',
+        sourceRange: scannedMixinReference.sourceRange!,
+      });
+      continue;
+    }
+    const mixin = mixins.values().next().value;
+    if (!(mixin instanceof PolymerElementMixin)) {
+      element.warnings.push({
+        message: `@mixes reference to a non-Mixin ${scannedMixinReference
```

Comment:
Add a second sentence with suggested fixes. e.g.

is it imported? does it have a `@polymerMixin` annotation?